In [10]:
import pandas as pd
import numpy as np

In [11]:
crimes_df = pd.read_csv('hate_crime.csv')

In [12]:
crimes_df

INCIDENT_ID  DATA_YEAR        ORI PUB_AGENCY_NAME PUB_AGENCY_UNIT  \
0              3015       1991  AR0040200          Rogers             NaN   
1              3016       1991  AR0290100            Hope             NaN   
2                43       1991  AR0350100      Pine Bluff             NaN   
3                44       1991  AR0350100      Pine Bluff             NaN   
4              3017       1991  AR0350100      Pine Bluff             NaN   
...             ...        ...        ...             ...             ...   
219572      1318026       2020  WY0170100        Sheridan             NaN   
219573      1142103       2020  WY0210100        Evanston             NaN   
219574      1318018       2020  WY0210100        Evanston             NaN   
219575      1318022       2020  WY0210100        Evanston             NaN   
219576      1318023       2020  WY0210100        Evanston             NaN   

       AGENCY_TYPE_NAME STATE_ABBR STATE_NAME       DIVISION_NAME REGION_NAME  \
0                  City         AR   Arkansas  West South Central       South   
1                  City         AR   Arkansas  West South Central       South   
2                  City         AR   Arkansas  West South Central       South   
3                  City         AR   Arkansas  West South Central       South   
4                  City         AR   Arkansas  West South Central       South   
...                 ...        ...        ...                 ...         ...   
219572             City         WY    Wyoming            Mountain        West   
219573             City         WY    Wyoming            Mountain        West   
219574             City         WY    Wyoming            Mountain        West   
219575             City         WY    Wyoming            Mountain        West   
219576             City         WY    Wyoming            Mountain        West   

        ...              OFFENDER_RACE      OFFENDER_ETHNICITY VICTIM_COUNT  \
0       ...                      White                     NaN            1   
1       ...  Black or African American                     NaN            1   
2       ...  Black or African American                     NaN            1   
3       ...  Black or African American                     NaN            2   
4       ...  Black or African American                     NaN            1   
...     ...                        ...                     ...          ...   
219572  ...                      White  Not Hispanic or Latino            2   
219573  ...                    Unknown                 Unknown            1   
219574  ...  Black or African American  Not Hispanic or Latino            1   
219575  ...                      White  Not Hispanic or Latino            1   
219576  ...                      White  Not Hispanic or Latino            1   

                                             OFFENSE_NAME  \
0                                            Intimidation   
1                                          Simple Assault   
2                                      Aggravated Assault   
3       Aggravated Assault;Destruction/Damage/Vandalis...   
4                                      Aggravated Assault   
...                                                   ...   
219572                                       Intimidation   
219573           Destruction/Damage/Vandalism of Property   
219574                                 Aggravated Assault   
219575                                     Simple Assault   
219576                                Theft From Building   

        TOTAL_INDIVIDUAL_VICTIMS                       LOCATION_NAME  \
0                            1.0  Highway/Road/Alley/Street/Sidewalk   
1                            1.0  Highway/Road/Alley/Street/Sidewalk   
2                            1.0                      Residence/Home   
3                            1.0  Highway/Road/Alley/Street/Sidewalk   
4                            1.0                 Service/Gas Station

In [13]:
crimes_df.columns

Index(['INCIDENT_ID', 'DATA_YEAR', 'ORI', 'PUB_AGENCY_NAME', 'PUB_AGENCY_UNIT',
       'AGENCY_TYPE_NAME', 'STATE_ABBR', 'STATE_NAME', 'DIVISION_NAME',
       'REGION_NAME', 'POPULATION_GROUP_CODE', 'POPULATION_GROUP_DESC',
       'INCIDENT_DATE', 'ADULT_VICTIM_COUNT', 'JUVENILE_VICTIM_COUNT',
       'TOTAL_OFFENDER_COUNT', 'ADULT_OFFENDER_COUNT',
       'JUVENILE_OFFENDER_COUNT', 'OFFENDER_RACE', 'OFFENDER_ETHNICITY',
       'VICTIM_COUNT', 'OFFENSE_NAME', 'TOTAL_INDIVIDUAL_VICTIMS',
       'LOCATION_NAME', 'BIAS_DESC', 'VICTIM_TYPES', 'MULTIPLE_OFFENSE',
       'MULTIPLE_BIAS'],
      dtype='object')

In [14]:
len(crimes_df)

219577

In [15]:
crimes_df['DATA_YEAR'].unique()

array([1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001,
       2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012,
       2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020])

In [16]:
crimes_df['STATE_ABBR'].unique()

array(['AR', 'AZ', 'CA', 'CO', 'CT', 'DC', 'DE', 'GA', 'IA', 'ID', 'IL',
       'KS', 'MA', 'MD', 'MN', 'MO', 'MS', 'NJ', 'NV', 'NY', 'OH', 'OK',
       'OR', 'PA', 'TN', 'TX', 'VA', 'WA', 'WI', 'AL', 'FL', 'IN', 'KY',
       'LA', 'ME', 'MI', 'NC', 'ND', 'RI', 'SC', 'UT', 'WY', 'AK', 'MT',
       'NM', 'SD', 'VT', 'NH', 'NB', 'WV', 'GM', 'FS', 'HI'], dtype=object)

In [17]:
crimes_df['STATE_NAME'].unique()

array(['Arkansas', 'Arizona', 'California', 'Colorado', 'Connecticut',
       'District of Columbia', 'Delaware', 'Georgia', 'Iowa', 'Idaho',
       'Illinois', 'Kansas', 'Massachusetts', 'Maryland', 'Minnesota',
       'Missouri', 'Mississippi', 'New Jersey', 'Nevada', 'New York',
       'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Tennessee', 'Texas',
       'Virginia', 'Washington', 'Wisconsin', 'Alabama', 'Florida',
       'Indiana', 'Kentucky', 'Louisiana', 'Maine', 'Michigan',
       'North Carolina', 'North Dakota', 'Rhode Island', 'South Carolina',
       'Utah', 'Wyoming', 'Alaska', 'Montana', 'New Mexico',
       'South Dakota', 'Vermont', 'New Hampshire', 'Nebraska',
       'West Virginia', 'Guam', 'Federal', 'Hawaii'], dtype=object)

In [18]:
#take out federal and Guam
crimes_df = crimes_df[crimes_df['STATE_ABBR'] != 'FS']
crimes_df = crimes_df[crimes_df['STATE_ABBR'] != 'GM']

In [19]:
#Check that df only consists of 50 states and DC
len(crimes_df['STATE_ABBR'].unique())

51

In [20]:
crimes_df['BIAS_DESC'].unique()
#Found "Anti-Asian", "Anti-Female"

array(['Anti-Black or African American', 'Anti-White', 'Anti-Jewish',
       'Anti-Arab', 'Anti-Protestant', 'Anti-Other Religion',
       'Anti-Islamic (Muslim)', 'Anti-Gay (Male)', 'Anti-Asian',
       'Anti-Catholic', 'Anti-Heterosexual', 'Anti-Hispanic or Latino',
       'Anti-Other Race/Ethnicity/Ancestry',
       'Anti-Multiple Religions, Group', 'Anti-Multiple Races, Group',
       'Anti-Lesbian (Female)',
       'Anti-Lesbian, Gay, Bisexual, or Transgender (Mixed Group)',
       'Anti-American Indian or Alaska Native',
       'Anti-Gay (Male);Anti-White',
       'Anti-Black or African American;Anti-Jewish',
       'Anti-Black or African American;Anti-Lesbian (Female)',
       'Anti-Black or African American;Anti-Gay (Male)',
       'Anti-Black or African American;Anti-White',
       'Anti-Atheism/Agnosticism', 'Anti-Gay (Male);Anti-Jewish',
       'Anti-Bisexual', 'Anti-Hispanic or Latino;Anti-White',
       'Anti-Hispanic or Latino;Anti-Multiple Races, Group',
       'Anti-Ame

In [21]:
crimes_df = crimes_df[crimes_df['BIAS_DESC'].str.contains('Anti-Asian')]

In [22]:
len(crimes_df)
#6462 hate crimes against Asians

6462

In [23]:
sum(crimes_df['BIAS_DESC'].str.contains('Anti-Female'))
#5 anti-asian, anti-female crimes

5

In [24]:
crimes_df['VICTIM_TYPES'].unique()

array(['Individual', 'Business;Individual', 'Other', 'Individual;Other',
       'Business', 'Society/Public', 'Government',
       'Individual;Society/Public', 'Religious Organization', 'Unknown',
       'Government;Individual', 'Individual;Religious Organization',
       'Financial Institution', 'Law Enforcement Officer',
       'Individual;Unknown', 'Individual;Law Enforcement Officer'],
      dtype=object)

In [25]:
anti_asian_df = crimes_df.groupby('DATA_YEAR', as_index = False).agg({'INCIDENT_ID': 'count'})

In [26]:
#Anti-Asian crimes per year where victims are an individual, a business/financial institution, a government entity, a religious organization, or society/public as a whole
anti_asian_df

DATA_YEAR  INCIDENT_ID
0        1991          269
1        1992          222
2        1993          260
3        1994          211
4        1995          355
5        1996          356
6        1997          346
7        1998          297
8        1999          304
9        2000          280
10       2001          280
11       2002          219
12       2003          236
13       2004          224
14       2005          208
15       2006          182
16       2007          190
17       2008          143
18       2009          128
19       2010          152
20       2011          137
21       2012          144
22       2013          134
23       2014          139
24       2015          115
25       2016          115
26       2017          132
27       2018          151
28       2019          199
29       2020          334

**From 2019 to 2020, there was a 68% increase in anti-Asian hate crimes. In the FBI's Hate Crime Statistics Program, victims are an individual, a business/financial institution, a government entity, a religious organization, or society/public as a whole.**

In [27]:
anti_asianfemale_df = crimes_df[crimes_df['BIAS_DESC'].str.contains('Anti-Female')]

In [28]:
anti_asianfemale_df.groupby('DATA_YEAR', as_index = False).agg({'INCIDENT_ID': 'count'})

DATA_YEAR  INCIDENT_ID
0       2016            1
1       2019            1
2       2020            3

**While we need more data on attacks against Asian females, the data indicates that there was a 200% increase in attacks directed toward this demographic.**

In [29]:
crimes_df

INCIDENT_ID  DATA_YEAR        ORI           PUB_AGENCY_NAME  \
43               33       1991  AZ0072300                   Phoenix   
55             3010       1991  AZ0072300                   Phoenix   
63               51       1991  CA0300900              Garden Grove   
71               77       1991  CO0010100                    Aurora   
116              95       1991  CO0031200      Chatfield State Park   
...             ...        ...        ...                       ...   
219258      1337705       2020  WASPD0000                   Seattle   
219319      1337899       2020  WASPD0000                   Seattle   
219342      1331508       2020  WI0130100                   Madison   
219359      1331485       2020  WI0130800  University of Wisconsin:   
219384      1268171       2020  WI0221400  University of Wisconsin:   

       PUB_AGENCY_UNIT       AGENCY_TYPE_NAME STATE_ABBR  STATE_NAME  \
43                 NaN                   City         AZ     Arizona   
55                 NaN                   City         AZ     Arizona   
63                 NaN                   City         CA  California   
71                 NaN                   City         CO    Colorado   
116                NaN     Other State Agency         CO    Colorado   
...                ...                    ...        ...         ...   
219258             NaN                   City         WA  Washington   
219319             NaN                   City         WA  Washington   
219342             NaN                   City         WI   Wisconsin   
219359         Madison  University or College         WI   Wisconsin   
219384     Platteville  University or College         WI   Wisconsin   

             DIVISION_NAME REGION_NAME  ... OFFENDER_RACE OFFENDER_ETHNICITY  \
43                Mountain        West  ...         White                NaN   
55                Mountain        West  ...       Unknown                NaN   
63                 Pacific        West  ...         White                NaN   
71                Mountain        West  ...         White                NaN   
116               Mountain        West  ...         White                NaN   
...                    ...         ...  ...           ...                ...   
219258             Pacific        West  ...       Unknown            Unknown   
219319             Pacific        West  ...       Unknown            Unknown   
219342  East North Central     Midwest  ...         White            Unknown   
219359  East North Central     Midwest  ...       Unknown            Unknown   
219384  East North Central     Midwest  ...       Unknown            Unknown   

       VICTIM_COUNT                              OFFENSE_NAME  \
43                1                              Intimidation   
55                1                              Intimidation   
63                2                        Aggravated Assault   
71                1                        Aggravated Assault   
116               1  Destruction/Damage/Vandalism of Property   
...             ...                                       ...   
219258            1                              Intimidation   
219319            1                              Intimidation   
219342            2                            Simple Assault   
219359            1  Destruction/Damage/Vandalism of Property   
219384            1                 Hacking/Computer Invasion   

        TOTAL_INDIVIDUAL_VICTIMS                       LOCATION_NAME  \
43                           1.0  Highway/Road/Alley/Street/Sidewalk   
55                           1.0                          Restaurant   
63                           2.0             Parking/Drop Lot/Garage   
71                           1.0                      Residence/Home   
116                          1.0                       Other/Unknown   
...                          ...                                 ...   
219258                       1.0            

In [30]:
states_2019_df = crimes_df[crimes_df['DATA_YEAR'] == 2019].groupby('STATE_NAME').agg({'INCIDENT_ID': 'count'})
states_2020_df = crimes_df[crimes_df['DATA_YEAR'] == 2020].groupby('STATE_NAME').agg({'INCIDENT_ID': 'count'})

In [31]:
states_2020_df.to_csv('states2020.csv', index = True)
!cp states2020.csv "drive/My Drive/"

cp: cannot create regular file 'drive/My Drive/': No such file or directory


In [32]:
states_2020_2019_df = states_2019_df.merge(states_2020_df, on = "STATE_NAME").rename(columns = {"INCIDENT_ID_x": "Anti-Asian crimes 2019", "INCIDENT_ID_y": "Anti-Asian crimes 2020"})

In [33]:
states_2020_2019_df['Percent increase between 2020 and 2019'] = ((states_2020_2019_df['Anti-Asian crimes 2020'] - states_2020_2019_df['Anti-Asian crimes 2019']) / states_2020_2019_df['Anti-Asian crimes 2019'])*100

In [34]:
states_2020_2019_df.sort_values(by = ['Percent increase between 2020 and 2019'], ascending = False, inplace = True)

In [35]:
states_2020_2019_df

Anti-Asian crimes 2019  Anti-Asian crimes 2020  \
STATE_NAME                                                             
Minnesota                                  1                       9   
New York                                   6                      30   
Nevada                                     2                       8   
Pennsylvania                               2                       8   
Arizona                                    1                       4   
Illinois                                   1                       3   
Texas                                      4                      11   
California                                43                      91   
Hawaii                                     2                       4   
Missouri                                   1                       2   
New Hampshire                              1                       2   
New Jersey                                39                      67   
Florida                                    2                       3   
Virginia                                   4                       5   
Colorado                                   6                       7   
District of Columbia                       1                       1   
Kentucky                                   2                       2   
Massachusetts                             13                      13   
Arkansas                                   1                       1   
Washington                                29                      26   
Ohio                                       6                       5   
Oregon                                     7                       5   
Mississippi                                3                       2   
North Carolina                             4                       2   
Idaho                                      2                       1   
Connecticut                                2                       1   
Michigan                                   6                       2   

                      Percent increase between 2020 and 2019  
STATE_NAME                                                    
Minnesota                                         800.000000  
New York                                          400.000000  
Nevada                                            300.000000  
Pennsylvania                                      300.000000  
Arizona                                           300.000000  
Illinois                                          200.000000  
Texas                                             175.000000  
California                                        111.627907  
Hawaii                                            100.000000  
Missouri                                          100.000000  
New Hampshire                                     100.000000  
New Jersey                                         71.794872  
Florida                                            50.000000  
Virginia                                           25.000000  
Colorado                                           16.666667  
District of Columbia                                0.000000  
Kentucky                                            0.000000  
Massachusetts                                       0.000000  
Arkansas                                            0.000000  
Washington                                        -10.344828  
Ohio                                              -16.666667  
Oregon                                            -28.571429  
Mississippi                                       -33.333333  
North Carolina                                    -50.000000  
Idaho                                             -50.000000  
Connecticut                                       -50.000000  
Michigan                                          -66.666667

In [36]:
from google.colab import drive
drive.mount('drive')

Mounted at drive


In [37]:
states_2020_2019_df.to_csv('states.csv', index = True)
!cp states.csv "drive/My Drive/"

In [38]:
states_2020_2019_df[states_2020_2019_df['Percent increase between 2020 and 2019'] > 100]

Anti-Asian crimes 2019  Anti-Asian crimes 2020  \
STATE_NAME                                                     
Minnesota                          1                       9   
New York                           6                      30   
Nevada                             2                       8   
Pennsylvania                       2                       8   
Arizona                            1                       4   
Illinois                           1                       3   
Texas                              4                      11   
California                        43                      91   

              Percent increase between 2020 and 2019  
STATE_NAME                                            
Minnesota                                 800.000000  
New York                                  400.000000  
Nevada                                    300.000000  
Pennsylvania                              300.000000  
Arizona                                   300.000000  
Illinois                                  200.000000  
Texas                                     175.000000  
California                                111.627907

**From 2019 to 2020, Minnesota and New York saw the greatest increase in anti-Asian crimes at 800% and 400%, respectively.**

In [39]:
crimes_df['LOCATION_NAME'].unique()

array(['Highway/Road/Alley/Street/Sidewalk', 'Restaurant',
       'Parking/Drop Lot/Garage', 'Residence/Home', 'Other/Unknown',
       'School/College', 'Grocery/Supermarket', 'Convenience Store',
       'Air/Bus/Train Terminal', 'Service/Gas Station', 'Field/Woods',
       'Liquor Store', 'Specialty Store',
       "Drug Store/Doctor's Office/Hospital", 'Bar/Nightclub',
       'Construction Site', 'Bank/Savings and Loan',
       'Commercial/Office Building', 'Hotel/Motel/Etc.',
       'Church/Synagogue/Temple/Mosque', 'Department/Discount Store',
       'Government/Public Building', 'Lake/Waterway/Beach',
       'Jail/Prison/Penitentiary/Corrections Facility',
       'Residence/Home;School/College', 'Rental Storage Facility',
       'Government/Public Building;Lake/Waterway/Beach',
       'Residence/Home;Restaurant',
       'Parking/Drop Lot/Garage;Residence/Home',
       'Highway/Road/Alley/Street/Sidewalk;Jail/Prison/Penitentiary/Corrections Facility',
       'Highway/Road/Alley/Stre

In [40]:
locations_2019_df = crimes_df[crimes_df['DATA_YEAR'] == 2019].groupby('LOCATION_NAME').agg({'INCIDENT_ID':'count'})
locations_2020_df = crimes_df[crimes_df['DATA_YEAR'] == 2020].groupby('LOCATION_NAME').agg({'INCIDENT_ID':'count'})

In [41]:
locations_2020_2019_df = locations_2019_df.merge(locations_2020_df, on = "LOCATION_NAME").rename(columns = {"INCIDENT_ID_x": "Anti-Asian crime location 2019", "INCIDENT_ID_y": "Anti-Asian crime location 2020"})

In [42]:
locations_2020_2019_df['Percent increase between 2020 and 2019'] = ((locations_2020_2019_df['Anti-Asian crime location 2020'] - locations_2020_2019_df['Anti-Asian crime location 2019']) / locations_2020_2019_df['Anti-Asian crime location 2019'])*100

In [43]:
locations_2020_2019_df.sort_values(by = ['Percent increase between 2020 and 2019'], ascending = False, inplace = True)

In [44]:
locations_2020_2019_df

Anti-Asian crime location 2019  \
LOCATION_NAME                                                                   
Convenience Store                                                           2   
Grocery/Supermarket                                                         1   
Other/Unknown                                                              11   
Park/Playground                                                             5   
Residence/Home                                                             35   
Hotel/Motel/Etc.                                                            2   
Service/Gas Station                                                         1   
Drug Store/Doctor's Office/Hospital                                         4   
Highway/Road/Alley/Street/Sidewalk                                         43   
School-College/University                                                   6   
Department/Discount Store                                                   3   
Air/Bus/Train Terminal                                                      5   
Church/Synagogue/Temple/Mosque                                              2   
Shopping Mall                                                               2   
Restaurant                                                                 14   
Parking/Drop Lot/Garage                                                    16   
Bank/Savings and Loan                                                       3   
Industrial Site                                                             1   
Field/Woods                                                                 1   
Cyberspace                                                                  1   
Specialty Store                                                             1   
Commercial/Office Building                                                  3   
Bar/Nightclub                                                               6   
School-Elementary/Secondary                                                15   
Government/Public Building                                                  7   
Jail/Prison/Penitentiary/Corrections Facility                               4   

                                               Anti-Asian crime location 2020  \
LOCATION_NAME                                                                   
Convenience Store                                                          13   
Grocery/Supermarket                                                         6   
Other/Unknown                                                              48   
Park/Playground                                                            15   
Residence/Home                                                             73   
Hotel/Motel/Etc.                                                            4   
Service/Gas Station                                                         2   
Drug Store/Doctor's Office/Hospital                                         8   
Highway/Road/Alley/Street/Sidewalk                                         72   
School-College/University                                                  10   
Department/Discount Store                                                   5   
Air/Bus/Train Terminal                                                      8   
Church/Synagogue/Temple/Mosque                                              3   
Shopping Mall                                                               3   
Restaurant                                                                 17   
Parking/Drop Lot/Garage                                                    18   
Bank/Savings and Loan                                                       3   
Industrial Site                                                             1   
Field/Woods                                                                 1   
Cyberspace                                                                  1

In [45]:
locations_2020_2019_df[locations_2020_2019_df['Percent increase between 2020 and 2019'] > 100]

Anti-Asian crime location 2019  \
LOCATION_NAME                                         
Convenience Store                                 2   
Grocery/Supermarket                               1   
Other/Unknown                                    11   
Park/Playground                                   5   
Residence/Home                                   35   

                     Anti-Asian crime location 2020  \
LOCATION_NAME                                         
Convenience Store                                13   
Grocery/Supermarket                               6   
Other/Unknown                                    48   
Park/Playground                                  15   
Residence/Home                                   73   

                     Percent increase between 2020 and 2019  
LOCATION_NAME                                                
Convenience Store                                550.000000  
Grocery/Supermarket                              500.000000  
Other/Unknown                                    336.363636  
Park/Playground                                  200.000000  
Residence/Home                                   108.571429

**From 2019 to 2020, there was a 550% and 500% increase in anti-Asian attacks at conveience stores and grocery/supermarkets, respectively.**

In [46]:
adult_juv_2019_df = crimes_df[crimes_df['DATA_YEAR'] == 2019].dropna(subset = ['ADULT_VICTIM_COUNT', 'JUVENILE_VICTIM_COUNT'])
adult_juv_2020_df = crimes_df[crimes_df['DATA_YEAR'] == 2020].dropna(subset = ['ADULT_VICTIM_COUNT', 'JUVENILE_VICTIM_COUNT'])

In [47]:
adult_2019 = sum(adult_juv_2019_df['ADULT_VICTIM_COUNT'])
adult_2020 = sum(adult_juv_2020_df['ADULT_VICTIM_COUNT'])

In [48]:
print("Number of Asian adult victims (18 or older) in 2019: ", adult_2019)
print("Number of Asian adult victims (18 or older) in 2020: ", adult_2020)
print("Percent increase in Asian adult victims: ", ((adult_2020 - adult_2019)/adult_2019)*100)

Number of Asian adult victims (18 or older) in 2019:  170.0
Number of Asian adult victims (18 or older) in 2020:  237.0
Percent increase in Asian adult victims:  39.411764705882355


In [49]:
juv_2019 = sum(adult_juv_2019_df['JUVENILE_VICTIM_COUNT'])
juv_2020 = sum(adult_juv_2020_df['JUVENILE_VICTIM_COUNT'])

In [50]:
print("Number of Asian juvenile victims (under 18) in 2019: ", juv_2019)
print("Number of Asian juvenile victims (under 18) in 2020: ", juv_2020)
print("Percent increase in Asian juvenile (under 18) victims: ", ((juv_2020 - juv_2019)/juv_2019)*100)

Number of Asian juvenile victims (under 18) in 2019:  10.0
Number of Asian juvenile victims (under 18) in 2020:  12.0
Percent increase in Asian juvenile (under 18) victims:  20.0


**From 2019 to 2020, hate crimes targeting Asian adults (18 or older) jumped nearly 40% while those targeting Asian juveniles (under 18) increased 20%.**